packages/agent/src/app.ts

```diff
@@ -265,10 +276,48 @@ export class App {
     }
   }
 
+  private getPingTestCommand(): string {
+    switch (platform()) {
+      case 'darwin':
+        return 'ping -c 1 -t 150 127.0.0.1';
+      case 'linux':
+        return 'ping -c 1 -w 150 127.0.0.1';
+      case 'win32':
+        return 'ping';
+      default:
+        throw new Error('Unsupported platform');
+    }
+  }
+
+  // This covers Windows, Linux, and Mac
+  private getPingCommand(ip: string, count = 1): string {
+    return platform() === 'win32' ? `ping /n ${count} ${ip}` : `ping -c ${count} ${ip}`;
+  }
+
+  private async isPingUtilAvailable(): Promise<boolean> {
+    if (this.pingUtilAvailable !== undefined) {
+      return Promise.resolve(this.pingUtilAvailable);
+    }
+    try {
+      await execAsync(this.getPingTestCommand(), { timeout: 200 });
+      this.pingUtilAvailable = true;
+    } catch (err: unknown) {
+      this.log.error(normalizeErrorString(err));
+      this.pingUtilAvailable = false;
+    }
+    return this.pingUtilAvailable;
+  }
+
   private async tryPingIp(message: AgentTransmitRequest): Promise<void> {
     try {
-      if (message.body && message.body !== 'PING') {
-        const warnMsg = 'Message body present but unused. Body should be empty for a ping request.';
+      if (!(await this.isPingUtilAvailable())) {
```

Comment:
What are the advantages of using this, vs just trying the ping and reporting the error?

I think https://github.com/medplum/medplum/issues/4007 would still be fixed by this PR, even without this check, just by virtue of the better error handling, yeah?

Reply:
Good point, removed those functions as they're kind of redundant